test-app/tests/proper-links/unit-test.ts

```diff
@@ -0,0 +1,33 @@
+import { module, test } from 'qunit';
+
+import { shouldHandle } from 'ember-primitives/proper-links';
+
+module('@properLinks', function () {
+  module('shouldHandle', function () {
+    test('hash-only links', async function (assert) {
+      function assertShouldHandle(
+        expected: boolean,
+        { location, href }: { location: string; href: string }
+      ) {
+        let url = new URL(location, 'https://example.com');
+
+        let anchor = document.createElement('a');
+
+        anchor.href = new URL(href, url).href;
+
+        let ignored: string[] = [];
+        let simpleClickEvent = new MouseEvent('click');
+
+        assert.strictEqual(shouldHandle(url.href, anchor, ignored, simpleClickEvent), expected);
+      }
+
+      assertShouldHandle(false, { location: '/foo', href: '/foo#bar' });
+      assertShouldHandle(false, { location: '/foo', href: '#bar' });
+      assertShouldHandle(false, { location: '/foo#bar', href: '/foo#other' });
+
+      assertShouldHandle(true, { location: '/foo', href: '/abc#xyz' });
+      assertShouldHandle(true, { location: '/foo#bar', href: '/abc#xyz' });
+      assertShouldHandle(true, { location: '/foo#bar', href: '/foo' });
```

Comment:
Not false, like line 24?

Reply:
In vanilla HTML, if you're on a page whose url has a fragment and you're linking to a path _without_ a fragment them the browser will do a full page refresh. This is true even if the path is the same and only the hash is changing

I don't think this behavior would be expected in a SPA.